lib/test.js

```diff
@@ -93,7 +93,22 @@ Test.prototype.run = function () {
         this.timeoutAfter(this._timeout);
     }
     this.emit('prerun');
-    this._cb(this);
+    var callbackReturn = this._cb(this);
+
+    if (
+        typeof Promise === 'function' &&
+        callbackReturn &&
+        typeof callbackReturn.then === 'function' &&
+        typeof callbackReturn.catch === 'function'
```

Comment:
.catch isn’t required; any thenable should be supported.

Reply:
Personally I only care about supporting async functions and not thenable returning functions.

However if you care strongly I can add support for thenable's